UI/volume-control.cpp

```diff
@@ -1153,8 +1153,13 @@ void VolumeMeter::paintVTicks(QPainter &painter, int x, int y, int height)
 
 inline int VolumeMeter::convertToInt(float number)
 {
+#if defined(_M_ARM64)
+	constexpr int min = (std::numeric_limits<int>::min)();
+	constexpr int max = (std::numeric_limits<int>::max)();
+#else
```

Comment:
What's the reason for this?

Reply:
This change was made to address compilation errors related to the use of `std::numeric_limits<int>::min()` and `std::numeric_limits<int>::max()`. The `()` are included to ensure that we are explicitly invoking the functions from the `<limits>` header rather than the macros that are defined.

This approach is derived from a previous [PR #5455](https://github.com/obsproject/obs-studio/pull/5455#discussion_r779210328) that addressed a similar Issue.

Reply:
It looks like our options are to `#define NOMINMAX` whenever doing `#include <Windows.h>` in C++ files (and making sure there are no issues), using this syntax all the time, or this (only using this syntax within an appropriate ifdef).

Reply:
Using `()` explicitly the code ensures clarity and avoids macro conflicts. This inline change only affects the particular line if broken, making it a localized fix.

Defining `NOMINMAX` affects the entire file, meaning any legitimate use of the `min` and `max` macros would be disabled throughout the file. This could lead to potential issues elsewhere in the code where the `min` and `max` macros might be needed.

Introducing conditional statements for this purpose adds redundancy and complexity to the code. It again adds `()` only in specific conditions, which can make the code harder to maintain and understand.

Reply:
I cannot think of legitimate uses of the `min` and `max` macros unless you're working with specific Windows headers that require them (such as perhaps GdiPlus headers, which we only use in two places).

The change shown in this review had already introduced conditional preprocessor statements here. The example you linked to unconditionally adds `()`. Unfortunately, this code preview was outdated, and I had not seen that the code was changed to unconditionally add `()`, which is what I was trying to hint at.

Reply:
@thirumalai-qcom, OBS Studio is a multi-platform application which relies on common API in C library API and C++ library API.

As it is, the `min` and `max` macro that are not part of the C/C++ APIs. And moreover it interfere with the C++ STL that is multi-platform.

Defining `NOMINMAX` is almost mandatory to make sure that the same C/C++ APIs is present on the 3 main platforms.

Those macros are not a thing on macOS or Linux, so disabling is simpler more sensible than making workaround for macros that OBS multi-platform code will never use.

Reply:
@RytoEX Could you share your thoughts on which approach to implement here? Should we continue using the `()` explicitly, or would it be more efficient to define `NOMINMAX` given the context of OBS Studio's multi-platform nature?

Reply:
- I have modified the PR by adding `NOMINMAX` flag to the 3 specific files where the macro conflicts arose.
- Adding `NOMINMAX` globally as suggested by @RytoEX breaks the compilation as some files already have it defined.

https://github.com/thirumalai-qcom/obs-studio/blob/ec5297b549aa2884287a1b7e986e0f2bf08ad143/UI/multitrack-video-output.hpp#L13